src/Admin/AmpThemes.php

```diff
@@ -0,0 +1,260 @@
+
+		return wp_parse_args( $theme, $default );
+	}
+
+	/**
+	 * Adds hooks.
+	 *
+	 * @return void
+	 */
+	public function register() {
+
+		add_filter( 'themes_api', [ $this, 'filter_themes_api' ], 10, 3 );
+
+		if ( ! wp_doing_ajax() && is_admin() ) {
+			add_action( 'current_screen', [ $this, 'register_hooks' ] );
+		}
+	}
+
+	/**
+	 * Register all hooks.
+	 *
+	 * @return void
+	 */
+	public function register_hooks() {
+
+		$screen = get_current_screen();
+
+		if ( $screen instanceof WP_Screen && in_array( $screen->id, [ 'themes', 'theme-install' ], true ) ) {
+			add_action( 'admin_enqueue_scripts', [ $this, 'enqueue_scripts' ] );
+		}
+	}
+
+	/**
+	 * Enqueue scripts and style for install theme screen.
+	 *
+	 * @return void
+	 */
+	public function enqueue_scripts() {
+
+		$asset_file   = AMP__DIR__ . '/assets/js/' . self::ASSET_HANDLE . '.asset.php';
+		$asset        = require $asset_file;
+		$dependencies = $asset['dependencies'];
+		$version      = $asset['version'];
+
+		wp_enqueue_script(
+			self::ASSET_HANDLE,
+			amp_get_asset_url( 'js/' . self::ASSET_HANDLE . '.js' ),
+			$dependencies,
+			$version,
+			true
+		);
+
+		wp_enqueue_style(
+			'amp-admin',
+			amp_get_asset_url( 'css/amp-admin.css' ),
+			[],
+			AMP__VERSION
+		);
+
+		$none_wporg = [];
+
+		$slugs = [];
+		foreach ( $this->get_themes() as $theme ) {
+			$slugs[] = $theme['slug'];
+			if ( ! isset( $theme['wporg'] ) || true !== $theme['wporg'] ) {
+				$none_wporg[] = $theme['slug'];
+			}
+		}
+
+		$js_data = [
+			'AMP_COMPATIBLE'    => self::AMP_COMPATIBLE,
+			'AMP_THEMES'        => $slugs,
+			'NONE_WPORG_THEMES' => $none_wporg,
+		];
+
+		wp_add_inline_script(
+			self::ASSET_HANDLE,
+			sprintf(
+				'var ampThemes = %s;',
+				wp_json_encode( $js_data )
+			),
+			'before'
+		);
+	}
+
+	/**
+	 * Filter the response of API call to wordpress.org for theme data.
+	 *
+	 * @param bool|object $response List of AMP compatible theme.
+	 * @param string      $action   API Action.
+	 * @param array       $args     Args for plugin list.
+	 *
+	 * @return object List of AMP compatible plugins.
+	 */
+	public function filter_themes_api( $response, $action, $args ) {
+
+		$args = (array) $args;
+		if ( ! isset( $args['browse'] ) || self::AMP_COMPATIBLE !== $args['browse'] ) {
+			return $response;
+		}
+
+		$response         = new stdClass();
+		$response->themes = [];
+
+		$args['per_page'] = ( ! empty( $args['per_page'] ) ) ? $args['per_page'] : 36;
+
+		$themes       = $this->get_themes();
+		$page         = ( ! empty( $args['page'] ) && 0 < (int) $args['page'] ) ? (int) $args['page'] : 1;
+		$theme_chunks = array_chunk( $themes, $args['per_page'] );
+		$themes       = ( ! empty( $theme_chunks[ $page - 1 ] ) && is_array( $theme_chunks[ $page - 1 ] ) ) ? $theme_chunks[ $page - 1 ] : [];
+
+		if ( 'query_themes' === $action ) {
+			foreach ( $themes as $i => $theme ) {
+				$response->themes[ $i ] = (object) $theme;
+			}
+		} else {
+			$response->themes = $themes;
```

Comment:
This doesn't seem to ever be called.

Reply:
Yes, We can remove that. But need to make sure the function only serve when $action is 'query_themes'.
I have updated it in the new PR #6681